### ISSUE TITLE
map.allowable_spill_position method has side effect on passed argument

### ISSUE THREAD

**Issue by tommasogallingani** [NOAA-ORR-ERD/PyGnome#144]
Hi,
I'm raising up the issue related to allowable_spill_position of GnomeMap object.
When using this method to check whether a position is allowable for spill the method has side effect on the passed argument (if an only if the arguments is a np.ndarray) and adds a dimension to the array with a in place modification of the variable.
This behavior might lead to errors in the model run in the spillability check is performed on Spill object start/end position.

tommaso

```
import os
import json
import gnome
import datetime
import numpy as np
import gnome.scripting as gs
from gnome.spills import PointLineRelease
from gnome.scripting import PyCurrentMover, PyWindMover
from gnome.movers import c_GridWindMover, c_GridCurrentMover
from gnome.spills.spill import _setup_spill as setup_spill_from_release

# setup model
model = gs.Model(
        start_time='2021-08-23T00:30:00',
        duration=getattr(gs, 'days')(9),
        time_step=gs.minutes(120)
    )

release_dict = {'num_elements': 1000,
 'start_position': np.array([35.92, 35.17, 0]),
 'end_position': np.array([35.92, 35.17, 0]),
 'release_time': '2021-08-23T10:00:00',
 'end_release_time': '2021-08-23T23:30:00'}
release = PointLineRelease(**release_dict)
arg_spill = {
             'amount': 1000,
             'units': 'kg',
             'windage_range': [0.01, 0.02],
             'windage_persist': -1
            }
spill = setup_spill_from_release(release=release,
                                 substance='oil_gas',
                                 name='spill',
                                 water=None,
                                 on=True,
                                 **arg_spill
                                 )

mymap = gs.GnomeMap(map_bounds=((34.041, 34.732), (34.041, 36.647), (36.327, 36.647), (36.327, 34.732)))
model.map = mymap

# testing tuple
position = (35.92,35.17, 0)
is_spillable = mymap.allowable_spill_position(position)
print(is_spillable,position)

# testing np.ndarray
position = np.array([35.92,35.17, 0])
is_spillable = mymap.allowable_spill_position(position)
print(is_spillable,position)
```

**Comment by jay-hennen** [comment]
That is an unfortunate side effect, which is fundamentally caused by allowable_spill_position not using the correct `point_in_poly` function, but the very closely related `points_in_poly` function.

Thanks for bringing this to our attention. A fix has been entered.

**Comment by ChrisBarker-NOAA** [comment]
Actually, it turns out that `points_in_poly()` is supposed to work with a single point, but it was a bug in that it altered the input array instead of making a copy -- in most (all) testing, we passed in a tuple, so it had to make a copy, so we didn't see the bug.

Thanks for the bug report.

@jay-hennen: the question for us is whether we ever need to pass multiple points into  `allowable_spill_position` -- it's documented as a single point, but maybe, for example, for a line spill, we might want more than one point.

'points_in_poly' has been fixed in case we do need that in the future.
